flink-libraries/flink-table/src/main/scala/org/apache/flink/table/codegen/CodeGenerator.scala

```diff
@@ -220,56 +247,105 @@ class CodeGenerator(
     // manual casting here
     val samHeader =
       // FlatMapFunction
-      if (clazz == classOf[FlatMapFunction[_,_]]) {
+      if (clazz == classOf[FlatMapFunction[_, _]]) {
```

Comment:
We can not choose `FlatMapFunction` or `RichFlatMapFunction` in `translateToPlan` method, unless We know whether the current rexNode contains `UserDefinedFunction`s. However CodeGenerator is a kind of RexVisitor and know `UserDefinedFunction` when `generateExpression` called. 
Besides, `RichFlatMapFunction` implements `FlatMapFunction`, so It's reasonable that Users call the generateFunction(desc, classOf[FlatMapFunction[Any, Any]], body, returnType) and returned a RichFlatMapFunction.